updated rules_mk_file to have addition support to .pgm files

VyshnaviKB opened the pull request:
Now user should be able to create the Bound RPGLE / SQLRPGLE programs with/without adding .pgm in the file name explicitly 
HELLO.PGM: HELLO.PGM.RPGLE
HELLOP.PGM: HELLOP.RPGLE
HELLOS.PGM: HELLOS.SQLRPGLE

<img width="1094" height="633" alt="Screenshot 2025-07-10 190246" src="https://github.com/user-attachments/assets/49d1dec4-9190-4771-ad0f-8fe44ffc5bfb" />
<img width="378" height="303" alt="Screenshot 2025-07-10 190300" src="https://github.com/user-attachments/assets/49e2151c-2663-4f43-b662-562b847f3c18" />
<img width="399" height="145" alt="Screenshot 2025-07-10 190317" src="https://github.com/user-attachments/assets/48c9a7fa-fdfa-41b8-a1f0-e8246a7b077f" />

irfanshar commented:
Great work! 

There's one more change that needs to be made. In the generated .Rules.mk.build (which gets cleaned in _post_make) we see that the RPGLE meant to be built into a PGM is still in the MODULEs list.

```
MODULEs := NEW.PGM
PGMs := HELLO.PGM


NEW.PGM_SRC=$(d)/HELLO.RPGLE
NEW.PGM_DEP=
NEW.PGM_RECIPE=PGM.RPGLE_TO_PGM_RECIPE
HELLO.PGM_SRC=$(d)/HELLO.PGM.RPGLE
HELLO.PGM_DEP=
HELLO.PGM_RECIPE=PGM.RPGLE_TO_PGM_RECIPE
```

VyshnaviKB commented:
So Irfan, here rules.mk file mapping is as below?
hello.pgm: hello.module

can you please confirm this?

irfanshar commented:
> So Irfan, here rules.mk file mapping is as below? hello.pgm: hello.module
> 
> can you please confirm this?

The following Rules.mk generated the above .Rules.mk.build

```
NEW.PGM: HELLO.RPGLE
HELLO.PGM: HELLO.PGM.RPGLE

```

irfanshar commented:
@VyshnaviKB Can you create an example for this in bob-recursive-example?

VyshnaviKB commented:
@irfanshar , i think we already have the some examples in bob-recursive repo.
for below scenarios 
**hello.rpgle, hello.sqlrpgle**
https://github.com/IBM/bob-recursive-example/blob/master/QRPGLESRC/ART301.SQLRPGLE
https://github.com/IBM/bob-recursive-example/blob/master/QRPGLESRC/LOG300.RPGLE

other than this above examples we need to add anything else?

edmundreinhardt commented:
> > @irfanshar , i think we already have the some examples in bob-recursive repo. for below scenarios **hello.rpgle, hello.sqlrpgle** https://github.com/IBM/bob-recursive-example/blob/master/QRPGLESRC/ART301.SQLRPGLE https://github.com/IBM/bob-recursive-example/blob/master/QRPGLESRC/LOG300.RPGLE
> > other than this above examples we need to add anything else?
> 
> I think we can just update the Rules.mk to showcase the new logic with the same RPGLE/SQLRPGLE source

Those are not examples because they are modules, they cannot be compiled into programs.
You would have to take some `foo.pgm.rpgle` and rename it to `foo.rpgle` and adjust the `rules.mk`

VyshnaviKB commented:
@irfanshar , added example in bob-recursive repo https://github.com/IBM/bob-recursive-example/pull/24
and updated the unit testcase as well, can you please review it
